### PR TITLE
Modify testing plan with example frameworks

### PR DESCRIPTION
**Pull request by Angela-CMU**

### CODE REVIEW

File: doc/design_doc.md [hunk 1]
```diff
@@ -32,10 +32,10 @@ To ensure the quality and the performance of the catalog implemented, a comprehe
...
+  * Metadata tests: We will focus on verifying the correct storage and retrieval of metadata. Tests will include different scenarios, including some edge cases. [Quickcheck](https://github.com/BurntSushi/quickcheck) is an example for performing the testing.
+  * [Documentation tests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#documentation-tests): Execute document examples 
 * Non-functional testing
-  * Performance evaluation: We will set up a performance baseline to compare the performance of our implementation. We can measure different metrics, for example, response time, throughput, etc.
+  * Microbenchmarking for performance evaluation: We can use [Criterion.rs](https://github.com/bheisler/criterion.rs?tab=readme-ov-file#features) and [bencher](https://github.com/bluss/bencher) to collect statistics to enable statistics-driven optimizations. In addition, we can set up a performance baseline to compare the performance with our implementation. We can measure different metrics, for example, response time, throughput, etc. Additionally,
```
Comment (yenjuw) [inline doc/design_doc.md]: remove "Additionally" in the end?

File: doc/design_doc.md [hunk 2]
```diff
@@ -32,10 +32,10 @@ To ensure the quality and the performance of the catalog implemented, a comprehe
 
 * Functional testing
   * API tests: For functional testing, we can achieve the goal through unit tests. We will test each API endpoint implemented in our project to ensure correct behavior. We will test various input parameters and validate the response format and the status code are as expected. Also, we will try to mimic possible edge cases and errors to ensure the implementation is robust and can perform suitable error handling. By doing so, we can ensure the API works as expected and provides correct results to clients. 
-  * Metadata tests: We will focus on verifying the correct storage and retrieval of metadata. Tests will include different scenarios, including some edge cases.
```
Comment (yenjuw) [inline doc/design_doc.md:35]: ａ

**Review by yenjuw — APPROVED** [review]
